### PR TITLE
Periodic Advertising Sync explicit scan and term fixes

### PR DESCRIPTION
**Pull request by Emil-Gydesen-Bose**
It is now possible to create a PA sync while explicitly scanning, and create PA sync in the termination (`term`) callback.

### CODE REVIEW

File: subsys/bluetooth/host/hci_core.c
Comment (cvinayak) [inline subsys/bluetooth/host/hci_core.c:4689]: I am assuming the fail is clear elsewhere now?

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/hci_core.c
Comment (Emil-Gydesen-Bose) [reply subsys/bluetooth/host/hci_core.c:4689]: `per_adv_sync_delete` (which replaces this lane), clears all flags

**Review by cvinayak — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
Looks fine to me, but would be good to have a review from @joerchan as well

**Comment by carlescufi** [comment]
> Looks fine to me, but would be good to have a review from @joerchan as well

He's away this week and we are nearing feature freeze, so to unblock @Emil-Gydesen-Bose 's work I'll merge this.

**Pull request merged**
